var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -452,7 +450,7 @@ def define_enable(suffix, value=None):
             define_trilinos_enable('Epetra'),
             define_trilinos_enable('EpetraExt'),
             define_trilinos_enable('FEI', False),
-            define_trilinos_enable('Gtest', False),
+            define_trilinos_enable('Gtest', self.run_tests),
```

Comment:
```suggestion
            define_trilinos_enable('Gtest', self.run_tests or spec.satisfies('dev_path=*')),
```

Reply:
Cool trick! Good to know how to detect `spack develop`.

Reply:
> Also let me run a couple tests to make sure I can get `nalu-wind` to run with it's own dependency on `gtest`. A few years ago there was an issue with an attempt to use our own direct dependency of `gtest` conflicting with `trilinos`'s version when that was tried on the project. If that is still the case then I will probably need to request that `gtest` stays on in `trilinos` until that can get sorted out.

I guess we'll have to restore a gtest variant, because I'm working on other projects where having gtest enabled *breaks* downstream projects: vendored dependencies are antithetical to large software ecosystems, so we should see if the tribits-installed gtest can use an external googletest if trilinos' gtest package has to stay enabled.

Reply:
> Cool trick! Good to know how to detect `spack develop`.

Yeah I'm thinking we should eventually add a function at the `package` level that just returns the result of this conditional.  something like `uses_tests()`.